chore: add pull request template (#1477)
apache/incubator-fury: .github/pull_request_template.md

<!--
**Thanks for contributing to Fury.**

**If this is your first time opening a PR on fury, you can refer to [CONTRIBUTING.md](https://github.com/apache/incubator-fury/blob/main/CONTRIBUTING.md).**

Contribution Checklist

    - The **Apache Fury (incubating)** community has restrictions on the naming of pr titles. You can also find instructions in [CONTRIBUTING.md](https://github.com/apache/incubator-fury/blob/main/CONTRIBUTING.md).

>     - Fury does not accept PRs that degrade performance. If the PR you submit will have an impact on performance, please Benchmark it yourself first.

Reviewer: ```suggestion
    - Fury has a strong focus on performance. If the PR you submit will have an impact on performance, please benchmark it first and provide the benchmark result here.
```
I personly feel `Fury does not accept PRs that degrade performance` is a little aggressive.

Author: Updated.

-->

## What does this PR do?

<!-- Describe the purpose of this PR. -->

## Related issues

<!--
Is there any related issue? Please attach here.

- #xxxx0
- #xxxx1
- #xxxx2
-->

## Does this PR introduce _any_ user-facing change?

<!--
If any user-facing interface changes, please [open an issue](https://github.com/apache/incubator-fury/issues/new/choose) describing the need to do so and update the document if necessary.
-->

> 

Reviewer: ```suggestion
- [ ] Does this PR introduce any public API change?
- [ ] Does this PR introduce any binary protocol compatibility change?
```

Author: Updated.

## Benchmark

<!--

> **Fury doesn't accept performance degradation**

Reviewer: Could we remove this line? The community should be open, this should be case by case.

Author: Updated.

<!--
**Thanks for contributing to Fury.**

**If this is your first time opening a PR on fury, you can refer to [CONTRIBUTING.md](https://github.com/apache/incubator-fury/blob/main/CONTRIBUTING.md).**

Contribution Checklist

    - The **Apache Fury (incubating)** community has restrictions on the naming of pr titles. You can also find instructions in [CONTRIBUTING.md](https://github.com/apache/incubator-fury/blob/main/CONTRIBUTING.md).

    - Fury has a strong focus on performance. If the PR you submit will have an impact on performance, please benchmark it first and provide the benchmark result here.
-->

## What does this PR do?

<!-- Describe the purpose of this PR. -->

## Related issues

<!--
Is there any related issue? Please attach here.

- #xxxx0
- #xxxx1
- #xxxx2
-->

> ## Does this PR introduce _any_ user-facing change?

Reviewer: ```suggestion
## Does this PR introduce any user-facing change?
```

Author: _any_(\_any\_) is in italics to emphasize importance, which I think is necessary

Reviewer: I prefer not emphasize it, the items list here are all important
